lib/chef/knife/bootstrap/options.rb

```diff
@@ -0,0 +1,356 @@
+              # the provided options to knife bootstrap, so we set the Chef::Config option here.
+            option :bootstrap_url,
+              long: "--bootstrap-url URL",
+              description: "URL to a custom installation script",
+              proc: Proc.new { |u| Chef::Config[:knife][:bootstrap_url] = u }
+
+              # bootstrap override: Do this instead of our own setup.sh from omnitruck. Causes bootstrap_url to be ignored.
+            option :bootstrap_install_command,
+              long: "--bootstrap-install-command COMMANDS",
+              description: "Custom command to install chef-client",
+              proc: Proc.new { |ic| Chef::Config[:knife][:bootstrap_install_command] = ic }
+
+              # bootstrap template: Run this command first in the bootstrap script
+            option :bootstrap_preinstall_command,
+              long: "--bootstrap-preinstall-command COMMANDS",
+              description: "Custom commands to run before installing chef-client",
+              proc: Proc.new { |preic| Chef::Config[:knife][:bootstrap_preinstall_command] = preic }
+
+              # bootstrap template
+            option :bootstrap_wget_options,
+              long: "--bootstrap-wget-options OPTIONS",
+              description: "Add options to wget when installing chef-client",
+              proc: Proc.new { |wo| Chef::Config[:knife][:bootstrap_wget_options] = wo }
+
+              # bootstrap template
+            option :bootstrap_curl_options,
+              long: "--bootstrap-curl-options OPTIONS",
+              description: "Add options to curl when install chef-client",
+              proc: Proc.new { |co| Chef::Config[:knife][:bootstrap_curl_options] = co }
+
+              # chef_vault_handler
+            option :bootstrap_vault_file,
+              long: "--bootstrap-vault-file VAULT_FILE",
+              description: "A JSON file with a list of vault(s) and item(s) to be updated"
+
+              # chef_vault_handler
+            option :bootstrap_vault_json,
+              long: "--bootstrap-vault-json VAULT_JSON",
+              description: "A JSON string with the vault(s) and item(s) to be updated"
+
+              # chef_vault_handler
+            option :bootstrap_vault_item,
+              long: "--bootstrap-vault-item VAULT_ITEM",
+              description: 'A single vault and item to update as "vault:item"',
+              proc: Proc.new { |i|
+                (vault, item) = i.split(/:/)
+                Chef::Config[:knife][:bootstrap_vault_item] ||= {}
+                Chef::Config[:knife][:bootstrap_vault_item][vault] ||= []
+                Chef::Config[:knife][:bootstrap_vault_item][vault].push(item)
+                Chef::Config[:knife][:bootstrap_vault_item]
+              }
+
+              # Windows only
+
+              # bootstrap template
+            option :install_as_service,
```

Comment:
We should also remove this since we've long ago deprecated this option in favor of windows task based installations

Reply:
Removed after discussion in community slack